### DISCUSSION TITLE
Feature Selection weights

### DISCUSSION THREAD

**Discussion by pimonteiro** [pycaret/pycaret#1490] [Q&A]
Hello,

I am having trouble finding out how to retrieve the selected features (using setup with feature_selection=True) weights. The source code shows a combination of two models (LightGBM and Random Forest), using correlation to decide which features should be selected. However, I can only retrieve the features selected and not their associated weight. I could train a model with these features and then collect each feature's weight, but it seems biased to me and not ideal. Is there any way to get the true weights from the selection?

**Comment by fawad-mahdi** [comment]
you're right. Unraveling the weights and features selected is pretty important for model deployment. This area is a bit black-boxed with pycaret, which I think can be sorted.
